source/extensions/filters/http/jwt_authn/filter.cc

```diff
@@ -119,7 +137,11 @@ void Filter::onComplete(const Status& status) {
         status == Status::JwtAudienceNotAllowed ? Http::Code::Forbidden : Http::Code::Unauthorized;
     // return failure reason as message body
     decoder_callbacks_->sendLocalReply(
-        code, ::google::jwt_verify::getStatusString(status), nullptr, absl::nullopt,
+        code, ::google::jwt_verify::getStatusString(status), 
+        [uri = this->original_uri_](Http::ResponseHeaderMap& headers) {
+          headers.setCopy(Http::LowerCaseString("www-authenticate"),
```

Comment:
Is it OK to send the same response for "token missing" case?

Reply:
So that's a tough one, it looks like from the [OAuth 2.0 RFC](https://tools.ietf.org/html/rfc6750) (which seems to be the most analogous to this situation) the `error` field should be included if the token is rejected but not if it's missing. If we think that we just need to follow the [401 Unauthorized RFC](https://tools.ietf.org/html/rfc7235#section-4.1), we just need `realm` to exist.

Reply:
By looking at OAuth 2.0 [RFC](https://tools.ietf.org/html/rfc6750),  for invalid token,  server SHOULD add `error=`.

How about:
 for missing we just add "realm=",  but for invalid token we also add error=. e.g.  "realm=URI, error=invalid_token"

Reply:
Agreed! Working on that change today